nexus/types/src/deployment/blueprint_diff.rs

```diff
@@ -577,40 +1100,69 @@ impl BpTableData for DiffDatasetsDetails {
 }
 
 #[derive(Debug)]
-pub struct ModifiedDataset {
-    pub before: BlueprintDatasetConfig,
-    pub after: BlueprintDatasetConfig,
-}
-
-#[derive(Debug)]
-pub struct BpDiffDatasetsModified {
-    pub generation_before: Option<Generation>,
-    pub generation_after: Option<Generation>,
-    pub datasets: Vec<ModifiedDataset>,
+pub struct BpDiffDatasetsModified<'a> {
+    pub generation_before: Generation,
+    pub generation_after: Generation,
+    pub datasets: Vec<&'a BlueprintDatasetConfigDiff<'a>>,
 }
 
-impl BpTableData for BpDiffDatasetsModified {
+impl BpTableData for BpDiffDatasetsModified<'_> {
     fn bp_generation(&self) -> BpGeneration {
         BpGeneration::Diff {
-            before: self.generation_before,
-            after: self.generation_after,
+            before: Some(self.generation_before),
+            after: Some(self.generation_after),
         }
     }
 
     fn rows(&self, state: BpDiffState) -> impl Iterator<Item = BpTableRow> {
-        self.datasets.iter().map(move |dataset| {
-            let before_strings = dataset.before.as_strings();
-            let after_strings = dataset.after.as_strings();
-
+        self.datasets.iter().map(move |&dataset| {
             let mut columns = vec![];
-            for (before, after) in std::iter::zip(before_strings, after_strings)
-            {
-                let column = if before != after {
-                    BpTableColumn::diff(before, after)
-                } else {
-                    BpTableColumn::value(before)
-                };
-                columns.push(column);
+
+            // Dataset Name
+            let before = DatasetName::new(
+                dataset.pool.before.clone(),
+                dataset.kind.before.clone(),
+            )
+            .full_name();
+            let after = DatasetName::new(
+                dataset.pool.after.clone(),
+                dataset.kind.after.clone(),
+            )
+            .full_name();
+            if before == after {
```

Comment:
Should this be a helper on `columns`? Not sure what to name it but something like

```rust
// internally compares and pushes either BpTableColumn::Value or BpTableColumn::Diff
columns.push_value_or_diff(before, after);
```

Reply:
Good idea. Except, `columns` is just a vec. However, I can create a constructor of `BpTableColumn` that always takes the before and after and does the comparison for us. I could even call it `new` 😄

Reply:
Hah, oops. Thanks, `new` looks great :sweat_smile: